src/test/java/net/snowflake/ingest/streaming/internal/ParameterProviderTest.java

```diff
@@ -291,8 +337,20 @@ public void testMaxChunksInBlobAndRegistrationRequest() {
     Properties prop = new Properties();
     Map<String, Object> parameterMap = getStartingParameterMap();
     parameterMap.put("max_chunks_in_blob_and_registration_request", 1);
-    ParameterProvider parameterProvider = new ParameterProvider(parameterMap, prop);
+    ParameterProvider parameterProvider = new ParameterProvider(parameterMap, prop, false);
+    Assert.assertEquals(1, parameterProvider.getMaxChunksInBlobAndRegistrationRequest());
+
+    parameterProvider = new ParameterProvider(parameterMap, prop, true);
     Assert.assertEquals(1, parameterProvider.getMaxChunksInBlobAndRegistrationRequest());
+
+    SFException e =
+        Assert.assertThrows(
+            SFException.class,
+            () -> {
+              parameterMap.put("max_chunks_in_blob_and_registration_request", 100);
+              new ParameterProvider(parameterMap, prop, true);
```

Comment:
curious, why shouldn't this test class also be made into a parameterized file so that all testcases test for the isIceberg=true mode of ParameterProvider too?

Reply:
Parameterized test.